consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -172,6 +231,16 @@ impl<T: Payload> BlockStore<T> {
             error!("fail to delete block: {:?}", e);
         }
         *self.inner.write().unwrap() = tree;
+        // Commit any blocks we can, this helps if we crash before state computer
+        // successfully persist the commits
```

Comment:
Can you please add more detail to what this helps with?  I'm a bit confused by the comment.

Reply:
it's also described in the pr, if we crash during commit B_n, after restart we'll start from B_i <- B_i+1 ... <- B_n and move on to process proposal/vote etc pretending we never commit B_n.

This helps to commit to B_n first and start processing proposal/vote.

Reply:
Thanks for the explanation.

Can you please update the comment to reflect this, it's not very intuitive.